qt/qt_common/map_widget.cpp

```diff
@@ -337,6 +344,68 @@ void MapWidget::initializeGL()
   ASSERT(m_contextFactory == nullptr, ());
   if (!m_screenshotMode)
     m_ratio = devicePixelRatio();
+
+#if defined(OMIM_OS_MAC)
+  m_apiOpenGLES3 = false;
+#elif defined(OMIM_OS_LINUX)
+  {
+    QOpenGLFunctions * funcs = context()->functions();
+    const std::string vendor = QString::fromLatin1(
+      (const char*)funcs->glGetString(GL_VENDOR)).toStdString();
+    const std::string renderer = QString::fromLatin1(
+      (const char*)funcs->glGetString(GL_RENDERER)).toStdString();
+    const std::string version = QString::fromLatin1(
+      (const char*)funcs->glGetString(GL_VERSION)).toStdString();
+    const std::string glslVersion = QString::fromLatin1(
+      (const char*)funcs->glGetString(GL_SHADING_LANGUAGE_VERSION)).toStdString();
+    const std::string extensions = QString::fromLatin1(
+      (const char*)funcs->glGetString(GL_EXTENSIONS)).toStdString();
+
+    LOG(LINFO, ("Vendor:", vendor, "\nRenderer:", renderer,
+                "\nVersion:", version, "\nShading language version:\n", glslVersion,
+                "\nExtensions:", extensions));
+
+    if (context()->isOpenGLES())
```

Comment:
Is this approach portable also to Windows?

Reply:
As portable as any other Qt functions. This is Qt code like the rest. The [doc doesn't state any restrictions.](https://doc.qt.io/qt-5/qopenglcontext.html#isOpenGLES)